matminer/featurizers/site.py

```diff
@@ -882,6 +882,173 @@ def implementors(self):
         return ['Qi Wang']
 
 
+class GaussianSymmFunc(BaseFeaturizer):
+    """
+    Gaussian symmetry function features suggested by Behler et al.,
+    based on pair distances and angles, to approximate the functional
+    dependence of local energies, originally used in the fitting of
+    machine-learning potentials.
+    The symmetry functions can be divided to a set of radial functions
+    (g2 function), and a set of angular functions (g4 function).
+    The number of symmetry functions returned are based on parameters
+    of etas_g2, etas_g4, zetas_g4 and gammas_g4.
+    See the original papers for more details:
+    “Atom-centered symmetry functions for constructing high-dimensional
+    neural network potentials”, J Behler, J Chem Phys 134, 074106 (2011).
+    The cutoff function is taken as the polynomial form (cosine_cutoff)
+    to give a smoothed truncation.
+    A Fortran and a different Python version can be found in the code
+    Amp: Atomistic Machine-learning Package
+    (https://bitbucket.org/andrewpeterson/amp).
+    Args:
+        etas_g2 (list of floats): etas used in radial functions.
+                                  (default: [0.05, 4., 20., 80.])
+        etas_g4 (list of floats): etas used in angular functions.
+                                  (default: [0.005])
+        zetas_g4 (list of floats): zetas used in angular functions.
+                                   (default: [1., 4.])
+        gammas_g4 (list of floats): gammas used in angular functions.
+                                    (default: [+1., -1.])
+        cutoff (float): cutoff distance. (default: 6.5)
+    """
+
+    @staticmethod
+    def cosine_cutoff(r, cutoff):
+        """
+        Polynomial cutoff function to give a smoothed truncation of the Gaussian
+        symmetry functions.
+        Args:
+            r (float): distance.
+            cutoff (float): cutoff distance.
+        Returns:
+            (float) cutoff function.
+        """
+        return 0 if r > cutoff else 0.5 * (np.cos(np.pi * r / cutoff) + 1.)
+
+    @classmethod
+    def g2(cls, eta, center_coord, neigh_coords, cutoff):
+        """
+        Gaussian radial symmetry function of the center atom,
+        given an eta parameter.
+        Args:
+            eta: radial function parameter.
+            center_coord (list of floats): coordinates of center atom.
+            neigh_coords (list of [floats]): coordinates of neighboring atoms.
+            cutoff (float): cutoff distance.
+        Returns:
+            (float) Gaussian radial symmetry function.
+        """
+        ridge = 0.
+        for neigh_coord in neigh_coords:
+            if str(neigh_coord) == str(center_coord):
+                continue
+            r = np.linalg.norm(neigh_coord - center_coord)
+            ridge += (np.exp(-eta * (r ** 2.) / (cutoff ** 2.)) *
+                      cls.cosine_cutoff(r, cutoff))
+        return ridge
+
+    @classmethod
```

Comment:
This can also be a staticmethod (not classmethod) - see above comment

Reply:
Thanks~~